### PR TITLE
tests: run integration tests against emulator

### PR DESCRIPTION
**Pull request by olavloite**
This will enable the integration tests to be executed against the emulator. This depends on the changes that are made in https://github.com/googleapis/java-spanner/pull/310, and so **this PR will not turn green until that PR has been merged** and a version of the Spanner client is available in the libraries-bom.

### CODE REVIEW

**Comment by skuruppu** [comment]
I'm happy to merge this in with the understanding that the tests against the emulator won't pass until the conditions you mentioned in your first comment are satisfied. Let me know what you think.

**Comment by olavloite** [comment]
> I'm happy to merge this in with the understanding that the tests against the emulator won't pass until the conditions you mentioned in your first comment are satisfied. Let me know what you think.

I would prefer waiting until the changes in the client library have been released. The changes that were needed for this PR are quite small and all in test files, so I don't think there's any real risk of (major) merge conflicts.

**Comment by skuruppu** [comment]
@olavloite, there is now a [new release](https://github.com/GoogleCloudPlatform/cloud-opensource-java/releases/tag/v8.1.0-bom) of the google-cloud-bom.

File: .github/workflows/integration-tests-against-emulator.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+on:
```
Comment (skuruppu) [inline .github/workflows/integration-tests-against-emulator.yaml:1]: If you can update this to reflect the new style in https://github.com/googleapis/java-spanner/blob/master/.github/workflows/integration-tests-against-emulator.yaml, that would be awesome.

File: .github/workflows/integration-tests-against-emulator.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (skuruppu) [inline .github/workflows/integration-tests-against-emulator.sh:1]: I think after the workflow config has been updated, you can delete this script.

File: .github/workflows/integration-tests-against-emulator.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  push:
+    branches:
+    - master
+  pull_request:
+name: integration-tests-against-emulator
```
Comment (skuruppu) [inline .github/workflows/integration-tests-against-emulator.yaml:6]: FYI: We recently found out that you can also give this a nicer name like "Integration Tests Against the Emulator".

File: .github/workflows/integration-tests-against-emulator.yaml (hunk 1 above)
Comment (olavloite) [reply .github/workflows/integration-tests-against-emulator.yaml:1]: Done.

File: .github/workflows/integration-tests-against-emulator.sh (hunk 2 above)
Comment (olavloite) [reply .github/workflows/integration-tests-against-emulator.sh]: Done.

File: .github/workflows/integration-tests-against-emulator.yaml (hunk 3 above)
Comment (olavloite) [reply .github/workflows/integration-tests-against-emulator.yaml]: Done.

**Pull request merged**